misc/vtk/src/processors/vtktovolume.cpp

```diff
@@ -0,0 +1,422 @@
+        const DataFormatBase *format =
+            vtkutil::getFormatFromVtkId(vtkTypeId, array->GetElementComponentSize() * 8);
+        if (format->getId() != selectedFormat) continue;
+        int numComps = array->GetNumberOfComponents();
+
+        numTotalComps += numComps;
+        dataArrays_.addArray(arrayName, numComps, array->GetVoidPointer(0));
+        if (keepSettings) {
+            BoolProperty *prop = dynamic_cast<BoolProperty *>(dataArrays_[dataArrays_.size() - 1]);
+            if (prop && formerChannels.find(prop->getDisplayName()) != formerChannels.end())
+                prop->set(true);
+        }
+    }
+    if (!keepSettings && numTotalComps <= 4)
+        for (auto *prop : dataArrays_.getProperties())
+            if (BoolProperty *boolProp = dynamic_cast<BoolProperty *>(prop)) boolProp->set(true);
+
+    dataArrays_.selectedFormat = dataFormats_.get();
+
+    if (keepSettings) convertData_ = true;
+
+    reactToChange_ = true;
+}
+
+void VTKtoVolume::updateAvailableArrays() {
+    NetworkLock lock;
+    reactToChange_ = false;
+
+    int numTotalComps = 0;
+    auto properties = dataArrays_.getProperties();
+    for (size_t p = 0; p < properties.size(); ++p) {
+        if (dynamic_cast<BoolProperty *>(properties[p])->get())
+            numTotalComps += dataArrays_.numArrayComponents[p];
+    }
+
+    for (size_t p = 0; p < properties.size(); ++p) {
+        if (dynamic_cast<BoolProperty *>(properties[p])->get()) continue;
+        bool tooMuch = numTotalComps + dataArrays_.numArrayComponents[p] > 4;
+        properties[p]->setReadOnly(tooMuch);
+    }
+
+    reactToChange_ = true;
+}
+
+void VTKtoVolume::convertData() {
+    outport_.setData(nullptr);
+    if (!inport_.hasData()) return;
+
+    auto vtkDataPtr = inport_.getData();
+
+    // Get dimensions. If none are given, it's not a regular grid.
+    std::optional<size3_t> dimensionsOpt = vtkDataPtr->getDimensions();
+    if (!dimensionsOpt) {
+        LogWarn("Input vtk data set is not rectilinear.");
+        return;
+    }
+    size3_t dimensions = *dimensionsOpt;
+    if (useCellData_.get())
+        for (size_t d = 0; d < 3; ++d) dimensions[d]--;
+
+    struct RawArrayData {
+        unsigned char *ptr;
+        size_t byteSize;
+    };
+    std::vector<RawArrayData> arrayDataPointers;
+
+    // How many components in total?
+    const DataFormatBase *format = DataFormatBase::get(dataFormats_.getSelectedValue());
+    size_t sizeElement = format->getSize();
+    size_t numTotalComps = 0;
+    for (size_t p = 0; p < dataArrays_.size(); ++p) {
+        BoolProperty *prop = dynamic_cast<BoolProperty *>(dataArrays_[p]);
+        if (prop && prop->get()) {
+            arrayDataPointers.push_back(
+                {dataArrays_.ptrArrayData[p], dataArrays_.numArrayComponents[p] * sizeElement});
+            numTotalComps += dataArrays_.numArrayComponents[p];
+        }
+    }
+    if (!numTotalComps) {
+        LogWarn("No components selected.");
```

Comment:
This should probably be stated when deselecting the last property. Should it even be allowed?

Reply:
Think it would be awkward, since  you'd get a warning while you're in the process of changing the setup, and probably plan to add arrays now anyway. But I will lock the button in that case.

Reply:
Since making a button read-only does not change anything, the button text is now changed to `Please Select Arrays` if no arrays are selected.